pkg/solana/logpoller/loader.go

```diff
@@ -0,0 +1,265 @@
+package logpoller
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink-common/pkg/services"
+)
+
+type Parser interface {
```

Comment:
There's an unfortunate naming conflict here we'll have to resolve at some point. One of the files in the evm log poller is parser.go and that component is referred to as the parser in code and docs. It parses DSL into SQL queries.

Not a big deal, but one or both we'll need to rename slightly to disambiguate.

Reply:
That makes sense to me. I can change it to whatever. Maybe `LogDataParser`?